src/object.rs

```diff
@@ -232,9 +232,73 @@ macro_rules! glib_object_wrapper {
         impl ::std::cmp::Eq for $name { }
     };
 
+    (@munch_impls $name:ident, ) => { };
+
+    (@munch_impls $name:ident, $super_name:path) => {
+        #[doc(hidden)]
+        impl<'a> $crate::translate::ToGlibPtr<'a,
+                *mut <$super_name as $crate::wrapper::Wrapper>::GlibType> for $name {
+            type Storage = <$crate::object::ObjectRef as
+                $crate::translate::ToGlibPtr<'a, *mut $crate::object::GObject>>::Storage;
+
+            #[inline]
+            fn to_glib_none(&'a self) -> $crate::translate::Stash<'a,
+                    *mut <$super_name as $crate::wrapper::Wrapper>::GlibType, Self> {
+                let stash = self.0.to_glib_none();
+                debug_assert!($crate::types::instance_of::<$super_name>(stash.0 as *const _));
+                $crate::translate::Stash(stash.0 as *mut _, stash.1)
+            }
+
+            #[inline]
+            fn to_glib_full(&self)
+                    -> *mut <$super_name as $crate::wrapper::Wrapper>::GlibType {
+                let ptr = self.0.to_glib_full();
+                debug_assert!($crate::types::instance_of::<$super_name>(ptr as *const _));
+                ptr as *mut _
+            }
+        }
+
+        impl $crate::object::IsA<$super_name> for $name { }
+    };
+
+    (@munch_impls $name:ident, $super_name:path => $super_ffi:path) => {
+        #[doc(hidden)]
+        impl<'a> $crate::translate::ToGlibPtr<'a, *mut $super_ffi> for $name {
+            type Storage = <$crate::object::ObjectRef as
+                $crate::translate::ToGlibPtr<'a, *mut $crate::object::GObject>>::Storage;
+
+            #[inline]
+            fn to_glib_none(&'a self) -> $crate::translate::Stash<'a, *mut $super_ffi, Self> {
+                let stash = self.0.to_glib_none();
+                debug_assert!($crate::types::instance_of::<$super_name>(stash.0 as *const _));
+                $crate::translate::Stash(stash.0 as *mut _, stash.1)
+            }
+
+            #[inline]
+            fn to_glib_full(&self) -> *mut $super_ffi {
+                let ptr = self.0.to_glib_full();
+                debug_assert!($crate::types::instance_of::<$super_name>(ptr as *const _));
+                ptr as *mut _
+            }
+        }
+
+        impl $crate::object::IsA<$super_name> for $name { }
+    };
+
+    (@munch_impls $name:ident, $super_name:path, $($implements:tt)*) => {
+        glib_object_wrapper!(@munch_impls $name, $super_name);
+        glib_object_wrapper!(@munch_impls $name, $($implements)*);
+    };
+
+    (@munch_impls $name:ident, $super_name:path => $super_ffi:path, $($implements:tt)*) => {
```

Comment:
I don't expect any ambiguity: in L288 `$super_name` is followed by `,` and here by `=>`.